packages/common/src/ProviderUtils.ts

```diff
@@ -60,6 +65,68 @@ export function createBasicProvider(nodeRetryConfig: RetryConfig[]): RetryProvid
   );
 }
 
+const KEY_TYPES = ["gckms", "mnemonic", "none"] as const;
+
+function getDefaultKeyType(): typeof KEY_TYPES[number] {
+  if (argv.network) {
+    const networkSplit = argv.network.split("_");
+    const keyType = networkSplit[networkSplit.length - 1];
+    if (KEY_TYPES.includes(keyType)) {
+      return keyType;
+    }
+  }
+  return "none";
+}
+
+function addMnemonicToProvider(
+  provider: AbstractProvider,
+  mnemonic: string = process.env.MNEMONIC ||
+    "candy maple cake sugar pudding cream honey rich smooth crumble sweet treat",
+  numKeys = process.env.NUM_KEYS ? parseInt(process.env.NUM_KEYS) : 2,
```

Comment:
would it be cleaner to do this:
```suggestion
  numKeys = parseInt(process.env.NUM_KEYS || 2),
```

Reply:
Oh actually, I don't think this compiles because parseInt doesn't take ints. Will convert to a string.